mcs/tools/mdoc/Mono.Documentation/monodocer.cs

```diff
@@ -165,35 +169,73 @@ class MDocUpdater : MDocCommand
 
 	HashSet<string> forwardedTypes = new HashSet<string> ();
 
-	public static string droppedNamespace = string.Empty;
+	public static DroppedAssemblyInfo.List droppedAssemblyList = new DroppedAssemblyInfo.List ();
+	public static string currentDroppedNamespace = string.Empty;
```

Comment:
Could we *please* not increase the amount of mutable static state?

`mdoc` is known to be slow; a "reasonable" way to improve this would be to make it multithreaded, which in theory sounds like it should be easy because each Type should be (reasonably) independent, so just fire off a thread per Type and be done, right?

But shared static state makes this impossible; we need to *decrease* the amount of static mutable state, not increase it! (Decreasing static mutable state also makes thinking about changes easier, as there are fewer side effects to consider.)